src/SocketTrebuchet.ts

```diff
@@ -19,21 +22,28 @@ const isPing = (data: Data) => {
   return buffer.length === 1 && buffer[0] === PING
 }
 
+const defaultDecode: Decode = (msg) => {
```

Comment:
We may compress the server payloads in the future, so i put the parsing of the mid in here.
Note that the client doesn't ever need to know the attempt, that is just for the server to keep track of.